builtin/builtin.ts

```diff
@@ -992,24 +993,38 @@ function matchesStructType(value: any, info: TypeInfo): boolean {
     return true
   }
 
-  if (info.fields && typeof value === 'object') {
-    // For struct type assertions, we need to check that the value has exactly
-    const descFields = Array.from(info.fields as Set<string>)
-    const valueFields = Object.keys(value)
+  if (info.methods && typeof value === 'object' && value !== null) {
+    const allMethodsMatch = Array.from(info.methods).every(
+      (method) => typeof value[method] === 'function'
+    )
+    if (allMethodsMatch) {
+      return true
+    }
+  }
 
-    const condition1 = descFields.every((field) => field in value)
-    const condition2 = valueFields.length === descFields.length
-    const condition3 = valueFields.every((field) => descFields.includes(field))
+  if (typeof value === 'object' && value !== null) {
+    const fieldNames = Object.keys(info.fields || {})
+    const valueFields = Object.keys(value)
 
+    const fieldsExist = fieldNames.every((field) => field in value)
+    const sameFieldCount = valueFields.length === fieldNames.length
+    const allFieldsInStruct = valueFields.every((field) => fieldNames.includes(field))
+    
     console.log('Struct field matching debug:', {
```

Comment:
remove this console.log

Reply:
matchesFunctionType(value: any, info: TypeInfo): boolean

info is not used, remove it.